ui/packages/consul-peerings/app/components/consul/peer/form/generate/index.hbs

```diff
@@ -47,6 +47,19 @@
       </fsm.State>
 
       <fsm.State @matches={{'success'}}>
+        {{yield (hash
+          Fieldsets=(component "consul/peer/form/token/fieldsets"
+            item=@item
+            token=fsm.state.context.PeeringToken
```

Comment:
who is assigning to fsm's state context? I don't see this being handled in the machine config for the generate form and no update to the configuration in this PR?

Reply:
Oh let me look I may have missed a file here 👀 , oh or maybe thats coming in the "wire all this together" PR, let me look and I'll come back to you

Reply:
This is set here I believe, as a result of the DataSource call:

https://github.com/hashicorp/consul/blob/main/ui/packages/consul-peerings/app/components/consul/peer/form/generate/index.hbs#L44

(just incase, I didn't miss a file here this is all good still)

Reply:
Does this mean you are ok to approve this PR?

I agree that this line would be better to use something else, probably event passing longer term, but its hidden under here so it's no big deal when we want to change that.

The interface we are using in https://github.com/hashicorp/consul/blob/main/ui/packages/consul-peerings/app/components/consul/peer/form/generate/index.hbs#L44 would probably stay the same.

Little note here, we are using `xstate/fsm` not `xstate`, so whilst its similar its not the same thing.

Reply:
Yes I'm ok with merging 

On a quick note - xstate/fsm pretty much works the same without nesting etc. assigning is the same for example